src/context.h

```diff
@@ -176,6 +176,7 @@ namespace librealsense
     // Helper functions for device list manipulation:
     std::vector<platform::uvc_device_info> filter_by_product(const std::vector<platform::uvc_device_info>& devices, const std::set<uint16_t>& pid_list);
     std::vector<std::pair<std::vector<platform::uvc_device_info>, std::vector<platform::hid_device_info>>> group_devices_and_hids_by_unique_id(
+        std::shared_ptr<context> ctx,
```

Comment:
I thought the serial num can be retrieved and  assigned during  platform::hid_device_info initialization, so passing ctx could be avoided

Reply:
the serial number is indeed retrieved and assigned during platform::hid_device_info in foreach_hid_device function
But we need to compare it to the UVC device serial number,  and to retrieve it we need the ctx because foreach_uvc_device
can't retrieve serial number when using MFEnumDeviceSources